### PR TITLE
feat: add kube_pod_container_status_last_terminated_timestamp metric

### PR DESCRIPTION
**Pull request by tetianakravchenko**
**What this PR does / why we need it**:
Adds `kube_pod_container_status_last_terminated_timestamp` metric to report the time when container was last time terminated with the `kube_pod_container_status_last_terminated_reason`. This metric is useful for cases like getting the crash situation - as described in this issue: https://github.com/kubernetes/kube-state-metrics/issues/1805#issuecomment-1219182114

**How does this change affect the cardinality of KSM**: *(increases, decreases or does not change cardinality)*
Adds a new metric in cases when the `kube_pod_container_status_last_terminated_reason` is reported

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Comment by rexagod** [comment]
@tetianakravchenko Tests seem to be failing here.

**Comment by tetianakravchenko** [comment]
Hey @rexagod @mrueg could you please check this PR?

File: internal/store/pod.go [hunk 1]
```diff
@@ -375,6 +376,32 @@ func createPodContainerStatusLastTerminatedExitCodeFamilyGenerator() generator.F
 	)
 }
 
+func createPodContainerStatusLastTerminatedTimestampFamilyGenerator() generator.FamilyGenerator {
+	return *generator.NewFamilyGeneratorWithStability(
+		"kube_pod_container_status_last_terminated_timestamp",
+		"Last terminated time for a pod container, when container was terminated with kube_pod_container_status_last_terminated_reason, in unix timestamp.",
```
Comment (rexagod) [inline internal/store/pod.go:382]:
```suggestion
		"Last terminated time for a pod container in unix timestamp.",
```
Is `when container was terminated with kube_pod_container_status_last_terminated_reason` necessary here?

**Review by rexagod — COMMENTED** [review]
This looks good, just a nit. :)

File: internal/store/pod.go (hunk 1 above)
Comment (tetianakravchenko) [reply internal/store/pod.go]: Hey @rexagod , thank you for the feedback! I've tried to be more precise in the description, but I think you are right, there is no need to keep it.
Cleaned it up in this commit - https://github.com/kubernetes/kube-state-metrics/pull/2291/commits/393430ed86941c502e179deffff625fee1864f3b, please have a look

**Comment by StephanErb** [comment]
This feature would be super helpful! Anything missing to get this in? :)

**Comment by tetianakravchenko** [comment]
Hey @rexagod could you please have another look on this PR?

**Comment by mrueg** [comment]
/lgtm 

Thanks for your contribution!

**Pull request merged**